### PR TITLE
Fixed empty `odo url list -o json`

### LINKED ISSUE #1515: odo url list -o json should return json with empty list if no url present
[kind/bug]

## What versions of software are you using?
- Operating System: macOS
- Output of `odo version`:

```
odo v0.0.20 (bd3e74a4)
Server: https://192.168.64.13:8443
Kubernetes: v1.11.0+d4cacc0
```

## How did you run odo exactly?
`odo url list --component <comp> --app <name> -o json`

## Actual behaviour
Currently it prints the text as mentioned below:
`✗  no URLs found for component front in application demo`

## Expected behaviour
If there are no URLs `odo url list --component <comp> --app <name> -o json` should return an empty list.
`
{
  "kind": "List",
  "apiVersion": "odo.openshift.io/v1alpha1",
  "metadata": {},
  "items": []
}
`

### PR DESCRIPTION
**Pull request by surajnarwade**
## What is the purpose of this change? What does it change?

returns empty json object if urls not present
## Was the change discussed in an issue?
fixes #1515

## How to test changes?

```
odo url list -o json
```

### CODE REVIEW

File: pkg/odo/cli/url/list.go [hunk 1]
```diff
@@ -53,17 +53,21 @@ func (o *URLListOptions) Run() (err error) {
...
-		if o.OutputFlag == "json" {
-			out, err := json.Marshal(urls)
-			if err != nil {
-				return err
-			}
-			fmt.Println(string(out))
+
+		if len(urls.Items) == 0 {
+
+			return fmt.Errorf("no URLs found for component %v in application %v", o.Component(), o.Application)
```
Comment (syamgk) [inline pkg/odo/cli/url/list.go]: i think here it should return an `empty url list` in json as well

File: pkg/odo/cli/url/list.go [hunk 2]
```diff
@@ -53,17 +53,21 @@ func (o *URLListOptions) Run() (err error) {
...
 	}
 
-	if len(urls.Items) == 0 {
-		return fmt.Errorf("no URLs found for component %v in application %v", o.Component(), o.Application)
+	if o.OutputFlag == "json" {
+		out, err := json.Marshal(urls)
+		if err != nil {
+			return err
+		}
+		fmt.Println(string(out))
```
Comment (syamgk) [inline pkg/odo/cli/url/list.go]: for me it returned, i think we should return an empty json in this case?
```
$ ./odo url list  -o json
 ✗  No component is set
```

Comment (surajnarwade) [reply pkg/odo/cli/url/list.go]: @syamgk , you are doing it from outside the context, this error is different, not related to `odo url list`

Comment (surajnarwade) [reply pkg/odo/cli/url/list.go]: Instead, try this `$ odo url list --context nodejs-ex/ -o json`

**Comment by surajnarwade** [comment]
@mohitsuman, can you please test this PR ?

File: pkg/odo/cli/url/list.go (hunk 2 above)
Comment (syamgk) [reply pkg/odo/cli/url/list.go]: yep

**Comment by surajnarwade** [comment]
@anmolbabu , I will add those tests once new flow for `odo url ` gets merged

**Comment by mohitsuman** [comment]
@surajnarwade Does this work with respect to new workflow ? or we can use it passing 
` --component <comp> --app <name>`

**Comment by surajnarwade** [comment]
@mohitsuman , we can use it passing
--component <comp> --app <name>

**Comment by mohitsuman** [comment]
> @mohitsuman , as per new workflow

@surajnarwade Then I think we can integrate it on vscode later only.

**Comment by surajnarwade** [comment]
@mohitsuman , can you check the desired output at least, we need to merge this

File: pkg/odo/cli/url/list.go [hunk 3]
```diff
@@ -96,5 +109,8 @@ func NewCmdURLList(name, fullName string) *cobra.Command {
 		},
 	}
 	genericclioptions.AddOutputFlag(urlListCmd)
+	urlListCmd.Flags().StringVar(&o.componentContext, "context", "", "Use context to indicate the path where the component settings need to be saved and this directory should contain component source for local and binary components")
```
Comment (kadel) [inline pkg/odo/cli/url/list.go]: please use `genericclioptions.AddContextFlag`

Comment (surajnarwade) [reply pkg/odo/cli/url/list.go]: fixed

File: tests/e2e/json_test.go [hunk 4]
```diff
@@ -29,6 +29,14 @@ var _ = Describe("odojsonoutput", func() {
 			runCmdShouldPass("odo create nodejs nodejs --app myapp --project json-test --git https://github.com/openshift/nodejs-ex")
 			runCmdShouldPass("odo push")
 		})
+		// odo url list -o json
+		It("should be able to list empty url list", func() {
```
Comment (amitkrout) [inline tests/e2e/json_test.go]: LGTM

**Pull request merged**